### PR TITLE
[NVD] Adapt the detection from the vendor feeds

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|#4859|

## Description

The purpose of this PR is to adapt the vulnerabilities collected from vendor feeds to the new structure:

- Save the vulnerabilities in the CVE hash table. This table contains only the information necessary to determine if a package is vulnerable or not.

- Iterate the hash table and send the alert reports. These reports have to be completed with the information that the feed has about the CVE.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h [hunk 1]
```diff
@@ -53,6 +53,8 @@ typedef enum vu_query {
     VU_REMOVE_UNUSED_ID,
     VU_JOIN_QUERY,
     VU_JOIN_RH_QUERY,
+    VU_JOIN_DEBIAN_QUERY,
```
Comment (chemamartinez) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h:56]: In this enumeration, one of these four is left over:

```
VU_JOIN_QUERY,
VU_JOIN_RH_QUERY,
VU_JOIN_DEBIAN_QUERY,
VU_GET_VULN_INFO,
```

according to the length of the `vu_queries[]` array.

Comment (chemamartinez) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h:56]: Forget it, I didn't see the last query.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -841,212 +1005,141 @@ int wm_vuldet_linux_rm_false_positivies(sqlite3 *db, agent_software *agent, OSHa
...
+        arch = (char *)sqlite3_column_text(stmt, 4);
+        operation = (char *)sqlite3_column_text(stmt, 5);
+        operation_value = (char *)sqlite3_column_text(stmt, 6);
+        pending = sqlite3_column_int(stmt, 7);
 
         if (!cve || !package || !version || !arch) {
             continue;
         }
 
+        if (!source) {
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1058]: This is necessary for Debian and Ubuntu systems. When the source is empty, it means that it is equal to the binary name.

[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1058]: On the other hand, RedHat systems always have the source empty. For them I also copy the name in the source so we can use it in the `cve_vuln_pkg` structure.

File: src/error_messages/debug_messages.h [hunk 3]
```diff
@@ -88,6 +88,8 @@
 #define VU_NO_HOTFIX_AVAIL    "(5577): It is not possible to perform a hotfix scan on agent %s."
 #define VU_NO_HOTFIX_FEED     "(5578): We have not found a hotfix that solves %s for agent %s in the Microsoft feed, so it is not possible to know it is vulnerable."
 #define VU_COMPARISSION_ERROR "(5579): The '%s' and '%s' versions of '%s' package could not be compared. Possible false positive."
+#define VU_PACKAGE_INSERT     "(5580): Agent (%s): Vulnerable package '%s' inserted into CVE '%s'."
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply src/error_messages/debug_messages.h]: Done!

**Pull request merged**